### PR TITLE
Add recommendation rule controller

### PR DESCRIPTION
**Pull request by qmhu**
#### What type of PR is this?

Feature

#### What this PR does / why we need it:

Add recommendation rule controller

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: pkg/known/label.go [hunk 1]
```diff
@@ -15,3 +15,9 @@ const (
 	AnalyticsUidLabel  = "analysis.crane.io/analytics-uid"
 	AnalyticsTypeLabel = "analysis.crane.io/analytics-type"
 )
+
+const (
```
Comment (xieydd) [inline pkg/known/label.go:19]: where we need those labels?

**Review by xieydd — CHANGES_REQUESTED** [review]

File: pkg/known/label.go (hunk 1 above)
Comment (qmhu) [reply pkg/known/label.go:19]: To know the recommendation from a RecommendationRule by label selector. for example ` kubectl crane` use it to get recommendation result.

Comment (xieydd) [reply pkg/known/label.go:19]: got it.

**Review by xieydd — CHANGES_REQUESTED** [review]

File: pkg/controller/recommendation/recommendation_rule_controller.go [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+			if time.Now().After(planingTime) {
+				currMissions = nil // reset missions to trigger creation for missions
+			}
+		}
+	}
+
+	if currMissions == nil {
+		// create recommendation missions for this round
+		for _, id := range identities {
+			currMissions = append(currMissions, analysisv1alph1.RecommendationMission{
```
[the remark this answers is not in the document]
Comment (qmhu) [reply pkg/controller/recommendation/recommendation_rule_controller.go:141]: one mission for one recommendation CR.

File: pkg/controller/recommendation/recommendation_rule_controller.go [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+
+		wg.Done()
+	}()
+
+	k := objRefKey(mission.TargetRef.Kind, mission.TargetRef.APIVersion, mission.TargetRef.Namespace, mission.TargetRef.Name)
+	if id, exist := identities[k]; !exist {
+		mission.Message = fmt.Sprintf("Failed to get identity, key %s. ", k)
+		return
+	} else {
+		recommendation := existingRecommendation
```
[the remark this answers is not in the document]
Comment (qmhu) [reply pkg/controller/recommendation/recommendation_rule_controller.go:372]: It's conventience to

File: pkg/controller/recommendation/recommendation_rule_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (xieydd) [reply pkg/controller/recommendation/recommendation_rule_controller.go:141]: ok, one mission for one object reference and one recommender.

**Review by xieydd — APPROVED** [review]

**Pull request merged**